DataDefinitions/StarSystem.cs

```diff
@@ -97,31 +97,43 @@ public void PreserveBodyData(List<Body> oldBodies, ImmutableList<Body> newBodies
             var newBodyBuilder = newBodies.ToBuilder();
             foreach (Body oldBody in oldBodies)
             {
-                int index = newBodyBuilder.FindIndex(b => b.bodyname == oldBody.bodyname);
-                if (oldBody.scanned != null)
+                if (newBodyBuilder.Any(b => b.bodyname == oldBody.bodyname))
```

Comment:
Probably not significant here, but we are in effect paying for the same search twice, here and in line 102.

Reply:
Agreed, they are similar. We use both because `FindIndex` is unable to distinguish between "not found" and "the first result"... both return an index of `0`.

Reply:
`FindIndex` returns -1 for not found.